cmd/main.go

```diff
@@ -265,3 +260,58 @@ func getOutputFn(outputFlag *string, currentFn selector.InstanceTypesOutputFn) s
 	}
 	return outputFn
 }
+
+func getRegionAndProfileAWSSession(regionName *string, profileName *string) *session.Session {
+	sessOpts := session.Options{}
+	if regionName != nil {
+		sessOpts.Config.Region = regionName
+	}
+
+	if profileName != nil {
+		if sessOpts.Config.Region == nil {
+			if profileRegion, err := getProfileRegion(*profileName); err != nil {
+				log.Println(err)
+			} else {
+				sessOpts.Config.Region = &profileRegion
+			}
+		}
+		sessOpts.Profile = *profileName
+	}
+
+	sess := session.Must(session.NewSessionWithOptions(sessOpts))
+	if sess.Config.Region != nil && *sess.Config.Region != "" {
+		return sess
+	}
+	if defaultProfileRegion, err := getProfileRegion(defaultProfile); err == nil {
+		sess.Config.Region = &defaultProfileRegion
+	}
+	if sess.Config.Region == nil || *sess.Config.Region == "" {
+		if defaultRegion, ok := os.LookupEnv(defaultRegionEnvVar); ok && defaultRegion != "" {
+			sess.Config.Region = &defaultRegion
+		}
+	}
```

Comment:
Should we add some logs in here when the resolution fails? Similar to line 273

Reply:
I didn't want to add warning in the other places because they're not required to be specified, so I didn't want a warning to come up all the time when someone only has their AWS_REGION env var set.

Reply:
I did add a new pretty long error message when no region could be found that specifies the places it looked.